src/charm.py

```diff
@@ -213,74 +146,70 @@ def get_manifest(self):
+            self._check_leader()
+        except CheckFailedError as err:
+            self.model.unit.status = err.status
+            return
 
-        Params:
-          manifest: (Optional) list of lightkube objects describing the entire application.  If
-                    omitted, will be computed using self.get_manifest()
-        """
-        if manifest:
-            expected_resources = manifest
-        else:
-            expected_resources = self.get_manifest()
-        found_resources = [None] * len(expected_resources)
-        errors = []
+        self.model.unit.status = MaintenanceStatus("Configuring dex charm")
+        self.ensure_state()
 
-        client = Client()
+        self._update_layer()
 
-        self.logger.info("Checking for expected resources")
-        for i, resource in enumerate(expected_resources):
-            try:
-                found_resources[i] = client.get(
-                    type(resource),
-                    resource.metadata.name,
-                    namespace=resource.metadata.namespace,
-                )
-            except ApiError:
-                errors.append(
-                    f"Cannot find k8s object for metadata '{resource.metadata}'"
-                )
-
-        self.logger.info("Checking readiness of found StatefulSets/Deployments")
-        statefulsets_ok, statefulsets_errors = validate_statefulsets_and_deployments(
-            found_resources
+        try:
+            self.handle_ingress()
+        except Exception as err:
+            self.model.unit.status = BlockedStatus(str(err))
+            return
+
+        self.model.unit.status = ActiveStatus()
+
+    def ensure_state(self):
+        self.state.set_default(
+            username="admin",
+            password="".join(choices(ascii_letters, k=30)),
+            salt=bcrypt.gensalt(),
+            user_id=str(uuid4()),
         )
-        errors.extend(statefulsets_errors)
 
-        # Log any errors
-        for err in errors:
-            self.logger.info(err)
+    def handle_ingress(self):
+        ingress = self._get_interface("ingress")
 
-        if len(errors) == 0:
-            return True
-        else:
-            raise CheckFailedError(
-                "Some Kubernetes resources missing/not ready.  See logs for details",
-                WaitingStatus,
-            )
+        if ingress:
+            for app_name, version in ingress.versions.items():
+                data = {
+                    "prefix": "/dex",
+                    "rewrite": "/dex",
+                    "service": self.model.app.name,
+                    "port": self.model.config["port"],
+                }
+
+                ingress.send_data(data, app_name)
+
+    def _check_leader(self):
+        if not self.unit.is_leader():
+            # We can't do anything useful when not the leader, so do nothing.
+            raise CheckFailedError("Waiting for leadership", WaitingStatus)
 
     def _get_interface(self, interface_name):
```

Comment:
This might need a refactor for the current usage.  If we get a NoVersionsListed, we set charm `WaitingStatus` but then return None and the charm continues to operate.  Is that the desired behaviour?  Looking at `handle_ingress()` I'm guessing this is not what we'd want, but I'm not sure